### PR TITLE
implemented watchlater and history functionality.Kindly review it and let me know what do you think about it !

### PR DESCRIPTION
**Pull request by kgajriaAgarwal**

### CODE REVIEW

File: src/Context/HistoryContext.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const encodedToken = localStorage.getItem("authData");
+    const [historyVideos , sethistoryVideos] = useState([]);
+    const {alertContent , setAlertContent} = useAlert();
+
+    //API - actionAddVideoToPlaylist
+    const addVideoToHistory = (videoObj) =>{
+        actionAddVideoToHistory({video:videoObj})
+      .then(res=> {
+          if(res.status === 201 || res.status === 200){
+            sethistoryVideos(res?.data?.history);
```
Comment (Adarsh-Patel-Dev) [inline src/Context/HistoryContext.jsx:22]: What is the use of  "?" after res and .data

**Comment by Adarsh-Patel-Dev** [comment]
Your watchlater & history feature is not working in hosted link

File: src/Context/HistoryContext.jsx (hunk 1 above)
Comment (Shivani-1524) [reply src/Context/HistoryContext.jsx:22]: i think it checks if the value is defined or not before proceeding its called optional chaining.

File: src/Pages/WatchLater/WatchLater.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        getAPI(getWatchlaterVideos(), "getWatchlaterVideos");
+      }, [watchLaterVideos]);
+
+
+    return(
+        <>
+            <p className='heading-md'>Watchlater...</p> 
+            <div className="col-12 flex-container-row watchlater-videos">
+    
+                {videoData?.length?
```
Comment (Shivani-1524) [inline src/Pages/WatchLater/WatchLater.jsx:41]: why name this cIndex and cVal ? i think the variable naming could be better unless you have a good reason for the c.

File: src/Shared/VideoCard/VideoCard.jsx [hunk 3]
```diff
@@ -1,29 +1,45 @@
...
  
     return(
         <>
-            
             <div className="card-container card-shadow video-card-container">
-                <button className="btn card-btn-dismiss" onClick={()=> {module==="LikedVideosPage"?removeFromLikedVideos(video_data._id) :addToLikedVideos(video_data)}}>
-                    {module==="LikedVideosPage"?<FcDislike  size='1.5rem' className=''/> :<FcLike size='1.5rem' className=''/>}
+                {/* LIKE / DISLIKE /REMOVE FROM HISTORY */}
+                <button className="btn card-btn-dismiss" 
+                    onClick={()=> {module==="LikedVideosPage"?
```
Comment (Shivani-1524) [inline src/Shared/VideoCard/VideoCard.jsx:32]: suggestion: i see that youve done chaining of the ternary here. it does reduce code readability from my pov.

File: src/Shared/VideoCard/VideoCard.jsx [hunk 4]
```diff
@@ -1,29 +1,45 @@
...
+                        removeFromLikedVideos(video_data._id) 
+                        : module==="historypage"? 
+                        removeVideoFromHistory(video_data._id)
+                        :addToLikedVideos(video_data)}
+                    }
+                >
+                    {module==="LikedVideosPage"?<FcDislike  size='1.5rem' className=''/> :
+                        module==="historypage"? 
+                        <MdCancel  size='1.5rem' className=''/> 
+                        :<FcLike size='1.5rem' className=''/>}
```
Comment (Shivani-1524) [inline src/Shared/VideoCard/VideoCard.jsx:42]: could remove the empty classNames here

**Review by Shivani-1524 — APPROVED** [review]
nice use of optional chaining and comments.

File: src/Shared/VideoCard/VideoCard.jsx (hunk 4 above)
Comment (kgajriaAgarwal) [reply src/Shared/VideoCard/VideoCard.jsx:42]: ok

**Pull request merged**